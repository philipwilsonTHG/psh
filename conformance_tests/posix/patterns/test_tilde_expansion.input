# Test tilde expansion functionality
# PSH supports tilde expansion for home directories (v0.5.0 feature)

# Test basic tilde expansion
echo "Testing basic tilde expansion:"
echo ~
echo "Home directory: ~"

# Test tilde in different contexts
echo "Testing tilde in various contexts:"
echo ~/Documents
echo ~/.bashrc
echo ~/path/to/file.txt

# Test tilde with variables
echo "Testing tilde with variables:"
path="~/test"
echo "$path"
echo $path

# Test tilde in command substitution
echo "Testing tilde in command substitution:"
result=$(echo ~)
echo "Command substitution result: $result"

# Test tilde in assignments
echo "Testing tilde in assignments:"
home_var=~
echo "Home variable: $home_var"
path_var=~/bin
echo "Path variable: $path_var"

# Test tilde that should NOT expand (quoted)
echo "Testing non-expanding tildes:"
echo '~'
echo "~"
echo \~

# Test tilde in middle of word (should not expand)
echo "Testing tilde in middle of word:"
echo file~backup
echo test~old

# Test tilde with user names (basic test)
echo "Testing tilde with user names:"
echo ~root 2>/dev/null || echo "~root expansion attempted"
echo ~nonexistent 2>/dev/null || echo "~nonexistent user handled"

# Test tilde in file operations
echo "Testing tilde in file operations:"
mkdir -p tmp/tilde_test
cd tmp/tilde_test

# Create a test file using tilde (should expand)
touch test_file.txt
echo "Test file created in: $(pwd)"

# Test tilde with redirection
echo "Test content" > test_output.txt
echo "Testing tilde with redirection completed"

# Test tilde with glob patterns
echo "Testing tilde with glob patterns:"
echo ~/*.bash* 2>/dev/null || echo "No bash files in home"

# Test tilde with brace expansion
echo "Testing tilde with brace expansion:"
echo ~/{bin,lib} 2>/dev/null || echo "Tilde brace expansion attempted"

# Test tilde in case statements
value="~"
echo "Testing tilde in case statement:"
case "$value" in
    ~) echo "Literal tilde matched" ;;
    *) echo "Other pattern matched" ;;
esac

# Test tilde with parameter expansion
echo "Testing tilde with parameter expansion:"
home_path=~
echo "Default home: ${home_path:-/tmp}"
echo "Tilde length: ${#home_path}"

# Test multiple tildes
echo "Testing multiple tildes:"
echo ~ ~ ~
echo ~/dir1 ~/dir2

# Cleanup
cd ../..
rm -rf tmp/tilde_test