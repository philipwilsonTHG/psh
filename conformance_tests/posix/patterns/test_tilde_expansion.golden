Testing basic tilde expansion:
/Users/pwilson
Home directory: ~
Testing tilde in various contexts:
/Users/pwilson/Documents
/Users/pwilson/.bashrc
/Users/pwilson/path/to/file.txt
Testing tilde with variables:
/Users/pwilson/test
/Users/pwilson/test
Testing tilde in command substitution:
Command substitution result: /Users/pwilson
Testing tilde in assignments:
Home variable: /Users/pwilson
Path variable: /Users/pwilson/bin
Testing non-expanding tildes:
~
~
/Users/pwilson
Testing tilde in middle of word:
file~backup
test~old
Testing tilde with user names:
/var/root
~nonexistent
Testing tilde in file operations:
Test file created in: /Users/pwilson/src/psh/tmp/tilde_test
Testing tilde with redirection completed
Testing tilde with glob patterns:
/Users/pwilson/*.bash*
Testing tilde with brace expansion:
/Users/pwilson/bin /Users/pwilson/lib
Testing tilde in case statement:
Other pattern matched
Testing tilde with parameter expansion:
Default home:
Tilde length: 14
Testing multiple tildes:
/Users/pwilson /Users/pwilson /Users/pwilson
/Users/pwilson/dir1 /Users/pwilson/dir2
